pkg/driver/node_server.go

```diff
@@ -1654,7 +1656,62 @@ func (driver *Driver) NodeGetVolumeStats(ctx context.Context, in *csi.NodeGetVol
 	log.Trace(">>>>> NodeGetVolumeStats")
 	defer log.Trace("<<<<< NodeGetVolumeStats")
 
-	return nil, status.Error(codes.Unimplemented, "")
+	volumePath := in.GetVolumePath()
+	if volumePath == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "volume path %s is empty", volumePath)
+	}
+
+	// check if it is a mount point
+	dummy := mount.New("")
+	notMount, err := dummy.IsLikelyNotMountPoint(volumePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, status.Errorf(codes.InvalidArgument, "volume path %s does not exist", volumePath)
+		}
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	if notMount {
+		return nil, status.Errorf(codes.InvalidArgument, "volume path %s is not mounted", volumePath)
+	}
+
+	folders := strings.Split(in.VolumePath, "/")
+	pvName := folders[len(folders)-2]
```

Comment:
nit : validate the size of folders before accessing by index

Reply:
done